src/sql/magic.py

```diff
@@ -639,6 +669,79 @@ def _persist_dataframe(
             table_name=table_name, data_frame=frame, if_exists=if_exists, index=index
         )
 
+    def get_query_type(self, command: SQLCommand, original: bool):
+        """
+        Returns the query type of the original sql command
+        """
+        to_parse = (
+            self.remove_leading_parentheses(command.sql_original)
+            if original
+            else self.remove_leading_parentheses(command.sql)
+        )
+        query_type = (
+            sqlparse.parse(to_parse)[0].get_type() if sqlparse.parse(to_parse) else None
+        )
+        return query_type
+
+    def remove_leading_parentheses(self, sql_command):
```

Comment:
this issue is supposed to close

https://github.com/ploomber/jupysql/issues/651
and
https://github.com/ploomber/jupysql/issues/652

is this parenthesis removing logic relevant to those? I'm a bit confused

Reply:
This is only needed when someone might use a CTE in duckdb in the following manner:

```
(WITH my_penguins as (
    SELECT * from penguins.csv
) SELECT * from my_penguins)
```

Line 406-408 where we check for is_cte would fail in the above case if there is no removing the parentheses.

Reply:
We discussed this parenthesis on a different issue, but I forgot the conclusion (I found this https://github.com/ploomber/jupysql/issues/879 but I think we discussed this somewhere else). In general, we try not to add new requirements to existing issues because it makes it harder to review the code.

also, I feel like we're mixing two separate issues here: CTE generation and SQL queries wrapped in parenthesis. these are two independent problems that we should tackle independently. so please remove this parenthesis handling logic and open an issue so we can discuss what we do about it

Reply:
I've removed it. Opening a new issue regarding the same.